go/pkg/marketplace/marketplace_service.go

```diff
@@ -169,20 +179,49 @@ func (s *MarkteplaceService) Collections(req *marketplacepb.CollectionsRequest,
 				SELECT * FROM trades t2 INNER JOIN activities_on_period aop
 				ON aop.id = t2.activity_id
 			),
+     activities_on_period_comparision AS (
```

Comment:
Did you test the performance of this query ? I think for now, when we do not find another optimized way for this query, just be sure that of related fields are indexed